### PR TITLE
[client/rest]: add NEM block routes

### PR DESCRIPTION
**Pull request by Wayonb**
problem: no nem block routes
solution: add nem block routes

### CODE REVIEW

File: client/rest/src/plugins/rosetta/nem/blockRoutes.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+		server.post('/block', rosettaPostRouteWithNetwork(blockchainDescriptor, BlockRequest, async typedRequest => {
+			const height = typedRequest.block_identifier.index;
+			const blockInfo = await services.proxy.fetch('local/block/at', undefined, {
+				method: 'POST',
+				body: JSON.stringify({ height }),
+				headers: { 'Content-Type': 'application/json' }
+			});
+
+			const parseTransaction = tx => parser.parseTransactionAsRosettaTransaction(tx.tx, { hash: { data: tx.hash } });
```
Comment (Jaguar0625) [inline client/rest/src/plugins/rosetta/nem/blockRoutes.js:55]: tx => transaction

Comment (Jaguar0625) [reply client/rest/src/plugins/rosetta/nem/blockRoutes.js:55]: any reason not to inline below?

File: client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+describe('NEM block routes', () => {
+	const assertRosettaErrorRaisedBasic = (...args) => assertRosettaErrorRaisedBasicWithRoutes(blockRoutes, ...args);
+	const assertRosettaSuccessBasic = (...args) => assertRosettaSuccessBasicWithRoutes(blockRoutes, ...args);
+
+	// region utils
+
+	const TEST_BLOCK_HEIGHT = 1111;
+	const TRANSACTION_HASH = 'A5DEA9FA1CB0C88EB9EC9156602C68902FB2F61E13342E552603DC55FF199D89';
+	const RECIPIENT_ADDRESS = 'TCYGRS7EWWHSNFUIETQIMB233NE75NVX6MJFI3JO';
+	const SIGNER_PUBLICKEY = '61127CD45073DFED58F472748900D0B90E2D8EC6E10EE4E41B03861D1D1A720D';
```
Comment (Jaguar0625) [inline client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js:49]: PUBLIC_KEY

File: client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+import { NemFacade } from 'symbol-sdk/nem';
+
+describe('NEM block routes', () => {
+	const assertRosettaErrorRaisedBasic = (...args) => assertRosettaErrorRaisedBasicWithRoutes(blockRoutes, ...args);
+	const assertRosettaSuccessBasic = (...args) => assertRosettaSuccessBasicWithRoutes(blockRoutes, ...args);
+
+	// region utils
+
+	const TEST_BLOCK_HEIGHT = 1111;
+	const TRANSACTION_HASH = 'A5DEA9FA1CB0C88EB9EC9156602C68902FB2F61E13342E552603DC55FF199D89';
```
Comment (Jaguar0625) [inline client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js:47]: consider lowercasing to test uppercasing

File: client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+				createTransferOperation(1, RECIPIENT_ADDRESS, '40000000', ...feeCurrencyProperties),
+				createTransferOperation(2, SIGNER_ADDRESS, '-2000', ...transferCurrencyProperties),
+				createTransferOperation(3, RECIPIENT_ADDRESS, '2000', ...transferCurrencyProperties),
+				createTransferOperation(4, SIGNER_ADDRESS, '-1000', ...feeCurrencyProperties)
+			]
+		);
+	};
+
+	const stubFetchNamespaceResult = (name, id, divisibility) => {
+		stubFetchResult(`namespace/definition/page?namespace=${id}&pageSize=100`, true, {
```
Comment (Jaguar0625) [inline client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js:116]: stubMosaicResolution

File: client/rest/src/plugins/rosetta/nem/blockRoutes.js (hunk 1 above)
Comment (Wayonb) [reply client/rest/src/plugins/rosetta/nem/blockRoutes.js:55]: It was just to reduce the line length but I will put it on multiple lines

File: client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js (hunk 4 above)
Comment (Jaguar0625) [reply client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js:118]: we can replace with test helper from other PR depending which is merged first 😅

File: client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+			// Act + Assert:
+			await assertRosettaErrorRaised(RosettaErrorFactory.INVALID_REQUEST_DATA, request => {
+				request.block_identifier.index = '2222';
+			});
+		});
+
+		it('fails when transaction is not found in cache', async () => {
+			// Arrange:
+			stubFetchResult(`transaction/get?hash=${TRANSACTION_HASH}`, true, createTransactionJson(1111, false));
```
Comment (Jaguar0625) [inline client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js:268]: TEST_BLOCK_HEIGHT ?

File: client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js (hunk 4 above)
Comment (Wayonb) [reply client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js:118]: Will wait until you checkin the other PRs

Comment (Jaguar0625) [reply client/rest/test/plugins/rosetta/nem/blockRoutes_spec.js:118]: actually, i have a few small cleanup, so i'll do it after this is merged

**Review by Jaguar0625 — APPROVED** [review]
👍

**Pull request merged**